cypress/integration/components/Molecules/SchoolLookup.spec.js

```diff
@@ -1,107 +1,124 @@
-                .find('table')
-                .should('contain', 'Prop name');
-        });
+    it('shows the table on button click', () => {
+      cy.get('@propsBtn').click();
+      cy.get('@container')
+        .find('table')
+        .should('contain', 'Prop name');
     });
+  });
 
-    describe('view code section', () => {
-        beforeEach(() => {
-            cy.get('[data-testid="SchoolLookup-example-1"]')
-                .as('container')
-                .find('[data-preview="SchoolLookup"]')
-                .as('preview');
+  describe('view code section', () => {
+    beforeEach(() => {
+      cy.get('[data-testid="SchoolLookup-example-1"]')
+        .as('container')
+        .find('[data-preview="SchoolLookup"]')
+        .as('preview');
 
-            cy.get('@container')
-                .find('button')
-                .contains('View Code')
-                .as('viewCodeBtn');
-        });
+      cy.get('@container')
+        .find('button')
+        .contains('View Code')
+        .as('viewCodeBtn');
+    });
 
-        it('has view code button', () => {
-            cy.get('@viewCodeBtn').should('exist');
-        });
+    it('has view code button', () => {
+      cy.get('@viewCodeBtn').should('exist');
+    });
 
-        it('shows code on click', () => {
-            cy.get('@viewCodeBtn').click();
-            cy.get('@container')
-                .find('textarea')
-                .should('exist');
-        });
+    it('shows code on click', () => {
+      cy.get('@viewCodeBtn').click();
+      cy.get('@container')
+        .find('textarea')
+        .should('exist');
+    });
 
-        it('closes code on click', () => {
-            cy.get('@viewCodeBtn').click();
-        });
+    it('closes code on click', () => {
+      cy.get('@viewCodeBtn').click();
     });
+  });
 
-    describe('School lookup functionality', () => {
-        beforeEach(() => {
-            cy.get('[data-testid="SchoolLookup-example-1"]')
-                .as('container')
-                .find('[data-preview="SchoolLookup"]')
-                .as('preview');
+  describe('School lookup functionality', () => {
+    beforeEach(() => {
+      cy.get('[data-testid="SchoolLookup-example-1"]')
+        .as('container')
+        .find('[data-preview="SchoolLookup"]')
+        .as('preview');
 
-            cy.get('@container')
-                .find('button')
-                .contains('View Code')
-                .as('viewCodeBtn');
-        });
+      cy.get('@container')
+        .find('button')
+        .contains('View Code')
+        .as('viewCodeBtn');
+    });
 
-        it('should verify schools lookup functionality', () => {
-            //verify error message for non school postcode
+    it('should verify schools lookup functionality', () => {
+      // Check if the API is up and running before running the tests.
+      cy.request({
+        url: 'https://lookups.sls.comicrelief.com/schools/lookup?query=NW5',
```

Comment:
Is there any reason we are using production lookup url here?

Reply:
Yeah. Need to use the same URL as the component. Which just uses the hardcoded prod URL. Not sure if it's possible to vary it depending on environment in the component library - depends if/how the components are bundled I guess.

Don't think there's any reason to use the staging url for this particular API as it's not creating anything